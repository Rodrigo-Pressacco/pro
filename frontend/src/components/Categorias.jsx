import React, { useState, useEffect } from "react";
import Categoria from './Categoria.jsx';
import "../styles/Categorias.css";
import axios from 'axios';
import url from "../utils/datosInicio.json"


const Categorias = (props) => {

    let categoriaFiltrada;
   
    
    
    const [datos, setDatos] = useState([null, false])
    // const [datosPorCategoria, setDatosPorCategoria] = useState()
    
    const handleClick = (nombre)=>{
        props.onSelect(nombre)
        categoriaFiltrada=nombre
        
        fetchAxios()
        
    }
    const fetchAxios =() =>{
        axios.get(`http://${url.hostUrl}:8080/productos/buscarPorCategoria/${categoriaFiltrada}`)
        .then( res => {
            const persons = res.data;
            props.onSelect2(persons)   
    })}
    
   
    /*Aca ya implemente para que la categoria se renderize desde la api y no modifique nada mas*/
    useEffect(()=>{
        axios.get(`http://${url.hostUrl}:8080/categorias/todos`)
        .then(res => {
            const persons = res.data;
            if (datos[1]===false) {
                setDatos([persons, true])                  
            }
        })
    })


   

    return (
        <div className="container">
            <div className="row">
                <h2>Buscar por tipo de alojamiento</h2>
                    <div className="col-md-4">
                    {datos[1] ? datos[0].map((e, index)=> (
                            <div onClick={()=>handleClick(e.titulo)} className="categoria" key={e.id}> 
                                <Categoria  nombre={e.titulo} image={e.url_imagen} cant={e.descripcion} />
                            </div>
                        )):<></>}
                    </div>
                        
            </div>
        </div>
    )
}

export default Categorias;
