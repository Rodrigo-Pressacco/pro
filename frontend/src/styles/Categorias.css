
@import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;500;700&display=swap');
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.15.3/css/all.min.css");

:root {
    --color-principal: #062f5b;
    --color-secundario: #679dd6;
    --color-texto: #5d596b;
}

* {
    font-family: 'Roboto', sans-serif;
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    
}

.col-md-4 {
    font-family: 'Roboto', sans-serif;
    font-weight: 300px;
    display: flex;
}

.col-md-4 img {
    border-radius: 20px 20px 0px 0px;
    width: 100%;
    height: 190px;
}

.categoria {
    border-radius: 20px;
    padding: 0px;
    width: 50%;
    border: 1px solid #F3F1ED;
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    margin: 20px 10px;
}

.categoria:hover{
    transform: scale(0.93);
    transition: all .2s ease-in-out; 
}

.container h2 {
    padding: 10px;
    font-size: 24px;
    color: #383B58;
    padding-bottom: 0px;
}

.col-md-4 p {
    margin-left: 10px;
    margin-bottom: 10px;
    font-size: 14px;
    color:#383B58;    
}



@media (max-width: 800px) {
    .col-md-4 {
        width: 100%;
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        flex-wrap: wrap;
        margin: auto;
    }

    .categoria{
        width: 47%;
        margin: 20px auto;
    }
}

@media (max-width: 485px) {
    .col-md-4 {
        width: 400px;
        flex-direction: row;
    }

    .categoria{
        width: 100%;
    }
}

