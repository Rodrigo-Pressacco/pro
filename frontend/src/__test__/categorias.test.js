import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Categorias from '../components/Categorias'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Categorias />', () => {
    let categorias
    let handleClick = jest.fn(); 
    let useEffect = jest.fn(); 

    beforeEach(() => {
        categorias = shallow(<Categorias handleClick={handleClick} useEffect={useEffect} />)
    });

    it("Deberia mostrar <Categorias/> correctamente", () => {
        expect(categorias).toMatchSnapshot();
    });

})