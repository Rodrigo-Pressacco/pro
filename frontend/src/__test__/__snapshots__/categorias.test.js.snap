// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categorias /> Deberia mostrar <Categorias/> correctamente 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h2>
      Buscar por tipo de alojamiento
    </h2>
    <div
      className="col-md-4"
    />
  </div>
</div>
`;
